OptimizelySDKCore/OptimizelySDKCoreTests/OptimizelyTest.m

```diff
@@ -382,6 +382,84 @@ - (void)testIsFeatureEnabledWithFeatureFlagEnabledAndUserIsBeingExperimented {
     [decisionServiceMock stopMocking];
 }
 
+// Should return false if the feature experiment variation’s `featureEnabled` property is true
+- (void)testIsFeatureEnabledWithVariationsFeatureEnabledFalse {
+    NSString *featureFlagKey = @"booleanFeature";
+    OPTLYFeatureFlag *featureFlag = [self.optimizely.config getFeatureFlagForKey:featureFlagKey];
+    OPTLYExperiment *experiment = [self.optimizely.config getExperimentForId:featureFlag.experimentIds[0]];
+    OPTLYVariation *variation = experiment.variations[0];
+    OPTLYFeatureDecision *decision = [[OPTLYFeatureDecision alloc] initWithExperiment:experiment variation:variation source:DecisionSourceExperiment];
+    
+    id decisionServiceMock = OCMPartialMock(self.optimizely.decisionService);
+    
+    OCMStub([decisionServiceMock getVariationForFeature:featureFlag userId:kUserId attributes:nil]).andReturn(decision);
+    
+    XCTAssertFalse([self.optimizely isFeatureEnabled:featureFlagKey userId:kUserId attributes:nil], @"should return false for disabled featureFlag");
+    
+    OCMVerify([decisionServiceMock getVariationForFeature:featureFlag userId:kUserId attributes:nil]);
+    [decisionServiceMock stopMocking];
+}
+
+// Should return true if the feature experiment variation’s `featureEnabled` property is true
+- (void)testIsFeatureEnabledWithVariationsFeatureEnabledtrue {
+    NSString *featureFlagKey = @"booleanFeature";
+    OPTLYFeatureFlag *featureFlag = [self.optimizely.config getFeatureFlagForKey:featureFlagKey];
+    OPTLYExperiment *experiment = [self.optimizely.config getExperimentForId:featureFlag.experimentIds[1]];
+    OPTLYVariation *variation = experiment.variations[1];
+    OPTLYFeatureDecision *decision = [[OPTLYFeatureDecision alloc] initWithExperiment:experiment variation:variation source:DecisionSourceExperiment];
+    
+    id decisionServiceMock = OCMPartialMock(self.optimizely.decisionService);
+    
+    OCMStub([decisionServiceMock getVariationForFeature:featureFlag userId:kUserId attributes:nil]).andReturn(decision);
+    
+    XCTAssertTrue([self.optimizely isFeatureEnabled:featureFlagKey userId:kUserId attributes:nil], @"should return true for enabled featureFlag");
+    
+    OCMVerify([decisionServiceMock getVariationForFeature:featureFlag userId:kUserId attributes:nil]);
+    
+    [decisionServiceMock stopMocking];
+}
+
+// Should return true if the user is bucketed into rollout experiment’s variation
+// and variation's featureEnabled is also true
```

Comment:
Comment on the code comment.  Our design doc requests:

Test that `isFeatureEnabled` returns true if the user is bucketed into the rollout experiment’s variation

I've also read in the design doc that "featureEnabled only appears in variations under experiments
which are feature experiments".  I infer from the design doc that rollout experiment variations will
not contain the key "featureEnabled".  It is the decision service response from `isFeatureEnabled`
which will be true.

Reply:
This was explicitly changed after @mikeng13 's [feedback](https://github.com/optimizely/ruby-sdk/pull/87/files/4e5f1b48c087bdff76eb0da0b48a357ec40b6ae4#diff-db360415c82dc1c3dd5a5c2872d0cd04R679).

Reply:
@abdurrafay is right. Rollout variations do contain the prop.

Reply:
I accept Mike Ng's response.